src/estimators.jl

```diff
@@ -472,8 +473,267 @@ function LGBMClassification(;
         categorical_feature, use_missing, linear_tree, feature_pre_filter, is_unbalance, boost_from_average, scale_pos_weight, sigmoid,
         drop_rate, max_drop, skip_drop, xgboost_dart_mode,
         uniform_drop, drop_seed, top_rate, other_rate, min_data_per_group, max_cat_threshold, cat_l2, cat_smooth,
-        metric, metric_freq, is_training_metric, ndcg_at, num_machines, local_listen_port, time_out,
+        metric, metric_freq, is_training_metric, eval_at, num_machines, local_listen_port, time_out,
         machine_list_file, num_class, device_type, gpu_use_dp, gpu_platform_id, gpu_device_id, num_gpu,
         force_col_wise, force_row_wise,
     )
 end
+
+mutable struct LGBMRanking <: LGBMEstimator
```

Comment:
I decided to create a separate ranking estimator (which has majority of params shared with the classifier plus some added new one for `lambdarank` and `rank_xendcg` objectives) as there might be some future changes that would be only applicable to the ranking tasks such as adding evaluation set and evaluation group query ids/sizes to the fit method.  The default objective has been set to `lambdarank` . This estimator is currently only available in the core LightGBM.jl, it's not added to the MLJ interface. This can be done in the future once it proves to work well.

Reply:
I decided to bring the `group_column` parameter since it's available in LightGBM but it does require the model to be stored as .txt which is not the usual use case as described above.